src/core/toga/widgets/canvas.py

```diff
@@ -1,20 +1,297 @@
+from contextlib import contextmanager
 from .base import Widget
+from math import pi
 
 
 class Canvas(Widget):
-    """"""
+    """Create new canvas
+
+    Args:
+        id (str):  An identifier for this widget.
+        style (:class:`colosseum.CSSNode`): An optional style object. If no
+            style is provided then a new one will be created for the widget.
+        factory (:obj:`module`): A python module that is capable to return a
+            implementation of this class with the same name. (optional &
+            normally not needed)
+    """
 
     def __init__(self, id=None, style=None, factory=None):
+        super().__init__(id=id, style=style, factory=factory)
+
+        # Create a platform specific implementation of Canvas
+        self._impl = self.factory.Canvas(interface=self)
+
+        self.rehint()
+
+    @contextmanager
+    def save_restore(self):
+        """Push contents to a stack, then restore to the previous state
+
+        Yields: None
+
         """
+        self._impl.save()
+        yield
+        self._impl.restore()
+
+    def draw(self, draw_func):
+        self._impl.draw(draw_func)
```

Comment:
If I draw without being inside a context manager, what is going to happen here? Would it make sense for every "draw directly on the canvas" operation to implicitly save and restore a context? Then, if you are *inside* a context manager, the creation of a new context can be skipped. Thoughts?